### PR TITLE
Split scipp-variable from scipp-core

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Part of #1047.

In principle this only moves code, but many changes related to different paths and namespaces required. Reviewers should mainly ensure that CI still runs all tests correctly and nothing got lost.

### CODE REVIEW

File: variable/include/scipp/variable/subspan_view.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#ifndef SCIPP_VARIABLE_SUBSPAN_VIEW_H
```
Comment (nvaytet) [inline variable/include/scipp/variable/subspan_view.h:5]: Why not `#pragma once`?

Comment (SimonHeybrock) [reply variable/include/scipp/variable/subspan_view.h:5]: I only changed this in files I was editing manually. We can do the rest of the include guards another time.

Comment (nvaytet) [reply variable/include/scipp/variable/subspan_view.h:5]: Yes, and it seemed this was a new file, so I thought you could change it here?

Comment (SimonHeybrock) [reply variable/include/scipp/variable/subspan_view.h:5]: For most files I just used `git mv`.

**Pull request merged**